nixos/modules/services/security/certmgr.nix

```diff
@@ -0,0 +1,210 @@
+    };
+
+    declSpecs = mkOption {
+      default = {};
+      example = literalExample ''
+      {
+        exampleCert =
+        let
+          domain = "example.com";
+          secret = name: "/var/lib/secrets/''${name}.pem";
+        in {
+          service = "nginx";
+          action = "reload";
+          authority = {
+            file.path = secret "ca";
+          };
+          certificate = {
+            path = secret domain;
+          };
+          private_key = {
+            owner = "root";
+            group = "root";
+            mode = "0600";
+            path = secret "''${domain}-key";
+          };
+          request = {
+            CN = domain;
+            hosts = [ "mail.''${domain}" "www.''${domain}" ];
+            key = {
+              algo = "rsa";
+              size = 2048;
+            };
+            names = {
+              O = "Example Organization";
+              C = "USA";
+            };
+          };
+        };
+      }
+      '';
+      type = with types; (attrsOf (submodule ({ ... }: {
+        options = {
+          service = mkOption {
+            type = nullOr str;
+            default = null;
+            description = "The service on which to perform &lt;action> after fetching.";
+          };
+
+          action = mkOption {
+            type = str;
+            default = "nop";
+            description = "The action to take after fetching.";
+          };
+
+          # These ought all to be specified according to certmgr spec def.
+          authority = mkOption {
+            type = attrs;
+            description = "certmgr spec authority object.";
+          };
+
+          certificate = mkOption {
+            type = nullOr attrs;
+            description = "certmgr spec certificate object.";
+          };
+
+          private_key = mkOption {
+            type = nullOr attrs;
+            description = "certmgr spec private_key object.";
+          };
+
+          request = mkOption {
+            type = nullOr attrs;
+            description = "certmgr spec request object.";
+          };
+        };
+    })));
+      description = ''
+        Certificate specs as described by:
+        <link xlink:href="https://github.com/cloudflare/certmgr#certificate-specs" />
+        These will be added to the Nix store, so they will be world readable.
+      '';
+    };
+
+    mkTargetDirs = mkOption {
+      default = false;
+      type = types.bool;
+      description = "Determines whether target directories (including parents) should be created for keys and certs in declarative specs.";
+    };
+
+    svcManager = mkOption {
+      default = "systemd";
+      type = types.enum [ "circus" "command" "dummy" "openrc" "systemd" "sysv" ];
+      description = ''
+        The behavior to use when executing actions after fetching certs
+        See: <link xlink:href="https://github.com/cloudflare/certmgr#certmgryaml" />.
```

Comment:
Whoops.. c/p error. as for the description. It is changed as part of https://github.com/NixOS/nixpkgs/pull/44556/commits/a94946e1505852aa7d8e0e7700af74e38c38819b.

Regarding other init-systems, yes. We probably won't need them in NixOS. But the `command` service manager is useful, when you just want to execute a verbatim command or script instead of invoking systemd, and `dummy` can be useful for debugging.

I'm personally ok with leaving the list as-is (vendor provided). But if you think so, I'm also happy to remove `circus` `openrc` and `sysv`. As long as default remains to be systemd.